py/torch_tensorrt/dynamo/compile.py

```diff
@@ -224,11 +232,27 @@ def compile_module(
             name=name,
         )
 
-        trt_modules[name] = trt_mod
+        # Get the engine and engine_str
+        assert trt_mod.engine
+
+        # Insert a call_function node to perform inference on TRT engine
+        with partitioned_module.graph.inserting_before(submodule_node):
+            new_node = partitioned_module.graph.call_function(
+                torch.ops.tensorrt.execute_engine,
+                (submodule_input_nodes, trt_mod.engine),
+            )
+
+        # Replace uses of submodule with the new_node
+        if submodule_node:
```

Comment:
Consider switching to `if submodule_node is not None`

Reply:
Additionally, should there ever be a case where this is `None`? Should an error be thrown if it is?